src/main/java/teammates/common/datatransfer/attributes/EntityAttributes.java

```diff
@@ -57,4 +58,18 @@ public boolean isValid() {
      * e.g. trim strings
      */
     public abstract void sanitizeForSaving();
+
+    /**
+     * Check the list for empty nodes ("") and remove them if found.
+     * Return the list without empty nodes or empty list if all nodes are empty
+     */
+    public List<String> removeEmptyElements(List<String> errors) {
+        Iterator<String> iterator = errors.iterator();
+        while (iterator.hasNext()) {
+            if (iterator.next().equals("")) {
+                iterator.remove();
+            }
+        }
+        return errors;
+    }
```

Comment:
Thanks, I refactored this as proposed, it really looks easier this way, and also learned from your comment, that it is not always good idea to modify arrays, which are received to method (they might need to be used after this method in another code). So I create new array in this method and return it.